lib/kitchen/directions/bake_numbered_exercise/v1.rb

```diff
@@ -2,15 +2,19 @@
 
 module Kitchen::Directions::BakeNumberedExercise
   class V1
-    def bake(exercise:, number:)
+    def bake(exercise:, number:, suppress_solution: false)
       problem = exercise.problem
       solution = exercise.solution
 
       problem_number = "<span class='os-number'>#{number}</span>"
 
       if solution.present?
-        problem_number = "<a class='os-number' href='##{exercise.id}-solution'>#{number}</a>"
-        bake_solution(exercise: exercise, number: number)
+        if !suppress_solution
+          problem_number = "<a class='os-number' href='##{exercise.id}-solution'>#{number}</a>"
+          bake_solution(exercise: exercise, number: number)
+        elsif suppress_solution
+          solution.trash
+        end
```

Comment:
what about

```ruby
if solution.present? and !suppress_solution
  # number & bake solution...
elsif suppress_solution
  # trash solution
end
```

Reply:
@kswanson33 It doesn't work, because for the `elsif` statement we also need to have solutions, to be able to trash them.
Here is the error:

![image](https://user-images.githubusercontent.com/40228854/116700791-75826d00-a9c7-11eb-9467-ee81d2ccca4c.png)

That's why I applied nested `if` statements.
I can change it to something like that:

![image](https://user-images.githubusercontent.com/40228854/116701151-ddd14e80-a9c7-11eb-8c57-97c7373f9b62.png)

if we want one `if` statement. Let me know what you think :wink:

Reply:
I like the second option, keeping it to one `if` statement is good! Another possibility is to use ruby's [safe navigation operator](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/). This would let you write `solution&.trash`.